src/caduceus/outboundSender.go

```diff
@@ -442,7 +471,7 @@ func (obs *CaduceusOutboundSender) dispatcher() {
 
 // worker is the routine that actually takes the queued messages and delivers
 // them to the listeners outside webpa
-func (obs *CaduceusOutboundSender) send(secret, acceptType string, msg *wrp.Message) {
+func (obs *CaduceusOutboundSender) send(urls *ring.Ring, secret, acceptType string, msg *wrp.Message) {
```

Comment:
`urls` is part of the `CaduceusOutboundSender`. We don't need to change the method signature.

Reply:
By sending the urls, we avoid the race condition where the `CaduceusOutboundSender` version could change, or needing to lock.  We don't want to lock if at all possible.